diff --git a/build/createIconSet.js b/build/createIconSet.js
index 15cd910ea81f08adcc8aefcc5ce597f2dbe316f0..5cd186351a405279d553f343037c4fa502ee20b2 100644
--- a/build/createIconSet.js
+++ b/build/createIconSet.js
@@ -26,11 +26,7 @@ export default function (glyphMap, fontName, expoAssetId, fontStyle) {
             componentWillUnmount() {
                 this._mounted = false;
             }
-            setNativeProps(props) {
-                if (this._icon) {
-                    this._icon.setNativeProps(props);
-                }
-            }
+            setNativeProps = this._icon?.setNativeProps;
             render() {
                 if (__DEV__ && this.props.name && !(this.props.name in glyphMap)) {
                     console.warn(`"${this.props.name}" is not a valid icon name for family "${fontName}"`);
