diff --git a/build/createIconSet.js b/build/createIconSet.js
index 7f655368b266af73c881673d72723a6a9a4d5391..e50ee6581330b85a52c64efedc185e4644176b84 100644
--- a/build/createIconSet.js
+++ b/build/createIconSet.js
@@ -31,11 +31,7 @@ export default function (glyphMap, fontName, expoAssetId, fontStyle) {
         componentWillUnmount() {
             this._mounted = false;
         }
-        setNativeProps(props) {
-            if (this._icon) {
-                this._icon.setNativeProps(props);
-            }
-        }
+        setNativeProps = this._icon?.setNativeProps;
         render() {
             if (__DEV__ && this.props.name && !(this.props.name in glyphMap)) {
                 console.warn(`"${this.props.name}" is not a valid icon name for family "${fontName}"`);
