diff --git a/lib/main.js b/lib/main.js
index 069f49c9e828771885e31cf507f414e5b06cfc85..569b1042ef0a3b75acdd85b114b54a2e19f47e14 100644
--- a/lib/main.js
+++ b/lib/main.js
@@ -90,7 +90,7 @@ function _parseVault (options) {
 }
 
 function _log (message) {
-  console.log(`[dotenv@${version}][INFO] ${message}`)
+  // console.log(`[dotenv@${version}][INFO] ${message}`)
 }
 
 function _warn (message) {
