diff --git a/_cjs/utils/siwe/createSiweMessage.js b/_cjs/utils/siwe/createSiweMessage.js
index 514170903a43c22087fa91aa5f6f904d4aaacc49..61e6e3d736ab56d0c60b60aa2c26999f12a8816c 100644
--- a/_cjs/utils/siwe/createSiweMessage.js
+++ b/_cjs/utils/siwe/createSiweMessage.js
@@ -118,7 +118,7 @@ function createSiweMessage(parameters) {
     return `${prefix}\n${suffix}`;
 }
 exports.createSiweMessage = createSiweMessage;
-const domainRegex = /^([a-zA-Z0-9][-a-zA-Z0-9]{0,61}[a-zA-Z0-9])\.[a-zA-Z]{2,}(:[0-9]{1,5})?$/;
+const domainRegex = /^([a-zA-Z0-9][-a-zA-Z0-9]{0,61}[a-zA-Z0-9]\.)+[a-zA-Z]{2,}(:[0-9]{1,5})?$/;
 const ipRegex = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(:[0-9]{1,5})?$/;
 const localhostRegex = /^localhost(:[0-9]{1,5})?$/;
 const nonceRegex = /^[a-zA-Z0-9]{8,}$/;
diff --git a/_esm/utils/siwe/createSiweMessage.js b/_esm/utils/siwe/createSiweMessage.js
index 62c50244c6ee635eef3a9f9b89a0754117235a36..2bd6b0ba7483b90ead4d57de61b6b478c4a7d4f5 100644
--- a/_esm/utils/siwe/createSiweMessage.js
+++ b/_esm/utils/siwe/createSiweMessage.js
@@ -133,7 +133,7 @@ export function createSiweMessage(parameters) {
     }
     return `${prefix}\n${suffix}`;
 }
-const domainRegex = /^([a-zA-Z0-9][-a-zA-Z0-9]{0,61}[a-zA-Z0-9])\.[a-zA-Z]{2,}(:[0-9]{1,5})?$/;
+const domainRegex = /^([a-zA-Z0-9][-a-zA-Z0-9]{0,61}[a-zA-Z0-9]\.)+[a-zA-Z]{2,}(:[0-9]{1,5})?$/;
 const ipRegex = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(:[0-9]{1,5})?$/;
 const localhostRegex = /^localhost(:[0-9]{1,5})?$/;
 const nonceRegex = /^[a-zA-Z0-9]{8,}$/;
diff --git a/utils/siwe/createSiweMessage.ts b/utils/siwe/createSiweMessage.ts
index 103d71c29505bdb118edd551b932587d498d8501..95e1e8fa030ea23beac4eee826bbe3bb1e66e58e 100644
--- a/utils/siwe/createSiweMessage.ts
+++ b/utils/siwe/createSiweMessage.ts
@@ -170,7 +170,7 @@ export function createSiweMessage(
 }
 
 const domainRegex =
-  /^([a-zA-Z0-9][-a-zA-Z0-9]{0,61}[a-zA-Z0-9])\.[a-zA-Z]{2,}(:[0-9]{1,5})?$/
+  /^([a-zA-Z0-9][-a-zA-Z0-9]{0,61}[a-zA-Z0-9]\.)+[a-zA-Z]{2,}(:[0-9]{1,5})?$/
 const ipRegex =
   /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(:[0-9]{1,5})?$/
 const localhostRegex = /^localhost(:[0-9]{1,5})?$/
